### PR TITLE
Add support for .gz ontologies

### PR DESCRIPTION
**Pull request by rctauber**

### CODE REVIEW

File: robot-core/src/main/java/org/obolibrary/robot/IOHelper.java [hunk 1]
```diff
@@ -335,34 +360,40 @@ public OWLOntology loadOntology(InputStream ontologyStream) throws IOException {
...
     try {
       OWLOntologyManager manager = OWLManager.createOWLOntologyManager();
-      manager.setIRIMappers(Sets.newHashSet(new CatalogXmlIRIMapper(catalogFile)));
-      ontology = manager.loadOntologyFromOntologyDocument(ontologyIRI);
+      // If a catalog file was loaded, set IRI mappers
+      if (catalogFile != null) {
+        manager.setIRIMappers(Sets.newHashSet(new CatalogXmlIRIMapper(catalogFile)));
+      }
+      // Maybe load a zipped ontology
+      if (ontologyIRI.toString().endsWith("gz")) {
```
Comment (jamesaoverton) [inline robot-core/src/main/java/org/obolibrary/robot/IOHelper.java]: I think this should look for `.gz`

File: robot-core/src/main/java/org/obolibrary/robot/IOHelper.java [hunk 2]
```diff
@@ -524,39 +559,19 @@ public OWLOntology saveOntology(
...
-      oboWriter.write(oboOntology, bw);
-    } else {
-      // use native save functionality
-      try {
-        ontology.getOWLOntologyManager().saveOntology(ontology, format, ontologyIRI);
-      } catch (OWLOntologyStorageException e) {
-        throw new IOException(String.format(ontologyStorageError, ontologyIRI.toString()), e);
-      }
+    // If saving in compressed format, get byte data then save to gzip
+    if (ontologyIRI.toString().endsWith("gz")) {
```
Comment (jamesaoverton) [inline robot-core/src/main/java/org/obolibrary/robot/IOHelper.java]: Should be `.gz`

**Comment by jamesaoverton** [comment]
I took a quick look. I'd like some sort of automated test, please.

**Comment by cmungall** [comment]
Will gz via http be supported?

**Comment by rctauber** [comment]
@cmungall - the current code includes support for `--input-iri` gz files
@jamesaoverton - I'll add a test for this.

**Comment by jamesaoverton** [comment]
This still needs a unit test, and there are two small changes that I commented on.

Please clarify in the docs that you can also save to `.gz`.

**Comment by rctauber** [comment]
Test added. There's info in `convert.md`, but I wasn't sure where else to put it. Any ideas?

**Comment by jamesaoverton** [comment]
Ok, the docs are fine.

I think the file extension checks should be `.endsWith(".gz")` throughout and not just `"gz"`. There are variants such as `".tgz"` that might not work properly with the current `"gz"` check.

**Comment by rctauber** [comment]
Makes sense - changed to `.gz`.

**Pull request merged**